Orange/widgets/unsupervised/owmds.py

```diff
@@ -920,69 +628,33 @@ def jitter(x, factor=1, rstate=None):
-            if self.label_only_selected:
-                if self._selection_mask is not None:
-                    for (x, y), text_item, selected \
-                            in zip(self.embedding, self._label_data,
-                                   self._selection_mask):
-                        if selected:
-                            self.plot.addItem(text_item)
-                            text_item.setPos(x, y)
-            else:
-                for (x, y), text_item in zip(self.embedding, self._label_data):
-                    self.plot.addItem(text_item)
-                    text_item.setPos(x, y)
-
-        self._legend_item = LegendItem()
-        viewbox = self.plot.getViewBox()
-        self._legend_item.setParentItem(self.plot.getViewBox())
-        self._legend_item.setZValue(viewbox.zValue() + 10)
-        self._legend_item.restoreAnchor(self.legend_anchor)
-
-        color_var = shape_var = None
-        color_index = self.cb_color_value.currentIndex()
-        if have_data and 1 <= color_index < len(self.colorvar_model):
-            color_var = self.colorvar_model[color_index]
-            assert isinstance(color_var, Orange.data.Variable)
-        shape_index = self.cb_shape_value.currentIndex()
-        if have_data and 1 <= shape_index < len(self.shapevar_model):
-            shape_var = self.shapevar_model[shape_index]
-            assert isinstance(shape_var, Orange.data.Variable)
-
-        if shape_var is not None or \
-                (color_var is not None and color_var.is_discrete):
-
-            legend_data = Mdsplotutils.legend_data(
-                color_var, shape_var, plotstyle=plotstyle)
-
-            for color, symbol, text in legend_data:
-                self._legend_item.addItem(
-                    ScatterPlotItem(pen=color, brush=color, symbol=symbol,
-                                    size=10),
-                    escape(text)
-                )
-        else:
-            self._legend_item.hide()
+            self.graph.plot_widget.addItem(curve)
+
+        coords = np.vstack((emb_x, emb_y)).T
+        attributes = () + self.data.domain.attributes + (self.variable_x, self.variable_y)
+        domain = Domain(attributes=attributes,
+                        class_vars=self.data.domain.class_vars,
+                        metas=self.data.domain.metas)
+        data = Table.from_numpy(domain, X=np.hstack((self.data.X, coords)),
+                                Y=self.data.Y,
+                                metas=self.data.metas)
+        self.graph.new_data(data, data[self._subset_mask])
+        connected_pairs()
+        self._update_data(self.variable_x, self.variable_y, True)
+
+    def _update_data(self, variable_x, variable_y, reset_view):
+        self.graph.update_data(variable_x, variable_y, reset_view=reset_view)
+        for axis in ["left", "bottom"]:
+            self.plot.hideAxis(axis)
+        self.plot.setAspectLocked(True, 1)
 
     def commit(self):
```

Comment:
This is not a new problem, but this PR is a good place to solve it since this code will be touched anyway.

What if the data already contains attributes "x" and "y" (or "mds-x", "mds-y", as suggested in another comment)? I think the function needs to be changed so that in this case it creates two new attributes with unique names (see  `Orange.widgets.utils.annotated_data.get_next_name`).